cmd/end2endtest/helpers.go

```diff
@@ -536,28 +537,43 @@ func (t *e2eElection) sendVotes(votes []*apiclient.VoteData) (errs map[int]error
 						delete(queue, i)
 					case errors.Is(err, context.DeadlineExceeded) || os.IsTimeout(err):
 						// if the context deadline is reached, no need to print it, just retry
-						timeouts.Add(1)
+						if timeoutsRetry.Load() > 5 {
+							errs = append(errs, err)
+							wg.Done()
+						}
+						timeoutsRetry.Add(1)
 					case strings.Contains(err.Error(), "mempool is full"):
 						log.Warn(err)
 						// wait and retry
-						_ = t.api.WaitUntilNextBlock()
-					case strings.Contains(err.Error(), "already exists") ||
-						strings.Contains(err.Error(), "overwrite count reached"):
+						waitErr := t.api.WaitUntilNextBlock()
```

Comment:
WaitUntilNextBlock will only fail (AFAIU) if the network is not producing blocks, a rare edge case. if the mempool is persisntently full, your code will still be stuck in an infinite loop. maybe you want to skip that `if waitErr != nil {` make waitErr count how many "mempool is full" errors happened? (possibly rename it to mempoolRetry then?)

Reply:
Yeah, in fact the code below I am counting how many "mempool is full" errors happened

`
						if waitErr != nil {
							if waitRetry.Load() > 5 {
								errs = append(errs, err)
								wg.Done()
							}
							waitRetry.Add(1)
}`

Reply:
updated to `mempoolRetry`